### PR TITLE
Trit 60: created new method for dynamic tax year and test cases

### PR DESCRIPTION
**Pull request by awasnik**
Trit 60: created new method for dynamic tax year and test cases

### CODE REVIEW

File: app/controllers/HomePageController.scala [hunk 1]
```diff
@@ -46,7 +46,8 @@ class HomePageController @Inject()(
     implicit request =>
 
       sessionRepository.set(UserAnswers(request.internalId)).map{ _ =>
-        Redirect(routes.HomePageController.onPageLoad())
+        Redirect(routes.RelevantTaxYearController.onPageLoad(NormalMode))
```
Comment (abhijeetardale) [inline app/controllers/HomePageController.scala]: Please check inject Navigator for redirection logic

File: conf/app.routes [hunk 2]
```diff
@@ -22,3 +22,11 @@ GET        /inconclusive-result                         controllers.Inconclusive
 GET        /non-tax-resident-result                     controllers.NonTaxResidentResultController.onPageLoad()
 
 GET        /tax-resident-result                         controllers.TaxResidentResultController.onPageLoad()
+
+GET        /select-tax-year                             controllers.RelevantTaxYearController.onPageLoad(mode: Mode = NormalMode)
+
+POST       /select-tax-year                             controllers.RelevantTaxYearController.onSubmit(mode: Mode = NormalMode)
+
+GET        /changeRelevantTaxYear                       controllers.RelevantTaxYearController.onPageLoad(mode: Mode = CheckMode)
```
Comment (abhijeetardale) [inline conf/app.routes]: please check the change urls, it should be "change-select-tax-year" for both GET and POST

File: conf/messages.en [hunk 3]
```diff
@@ -170,3 +170,12 @@ taxResidentResult.heading=Resident
 taxResidentResult.forTaxYear=For the tax year
 taxResidentResult.taxYear=2018-19
 taxResidentResult.info=This information is based on the details you have provided and is for information only.
+
+relevantTaxYear.title=Which tax year do you want to check your tax residency status for
+relevantTaxYear.heading=Which tax year do you want to check your tax residency status for?
+relevantTaxYear.pY=2018-19
```
Comment (abhijeetardale) [inline conf/messages.en]: removed the used messages

File: test/views/RelevantTaxYearSpec.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  )
+
+  val html: Html = renderer.render("relevantTaxYear.njk" ,json).futureValue
+
+  behave like normalPage(html, "relevantTaxYear",
+    "pY", "pY-1", "pY-2","pY-3")
+
+  behave like optionsPage[RelevantTaxYear](
+    renderer.render, "relevantTaxYear.njk", form, "value", RelevantTaxYear.radios)
+
```
Comment (abhijeetardale) [inline test/views/RelevantTaxYearSpec.scala]: could you please check error page behavior?

File: conf/app.routes (hunk 2 above)
Comment (awasnik) [reply conf/app.routes]: fixed

File: conf/messages.en (hunk 3 above)
Comment (awasnik) [reply conf/messages.en]: fixed

File: test/views/RelevantTaxYearSpec.scala (hunk 4 above)
Comment (awasnik) [reply test/views/RelevantTaxYearSpec.scala]: fixed

**Pull request merged**